app/src/main/java/net/gsantner/markor/activity/DocumentActivity.java

```diff
@@ -88,17 +90,22 @@ private static void launch(
         final AppSettings as = ApplicationObject.settings();
 
         final Intent intent;
-        if (GsFileBrowserListAdapter.isVirtualFolder(file) || file.isDirectory()) {
+        if (GsFileUtils.isDirectory(file)) {
             intent = new Intent(activity, MainActivity.class);
-            intent.addFlags(Intent.FLAG_ACTIVITY_BROUGHT_TO_FRONT);
         } else {
             intent = new Intent(activity, DocumentActivity.class);
 
-            if (!(activity instanceof DocumentActivity) &&
-                    Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP &&
-                    as.isMultiWindowEnabled()
-            ) {
+            final boolean lollipop = Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP;
+            final boolean fromDocumentActivity = activity instanceof DocumentActivity;
+            final boolean isMultiWindow = as.isMultiWindowEnabled();
```

Comment:
New behavior of launching DocumentActivity when one is already open is as follows:

1. Launch into existing Activity.
2. If the requested document matches current document, jump to requested line number
3. Else open a new document